spec/set-prototype-is-subset-of.html

```diff
@@ -2,18 +2,18 @@
 
 <emu-alg>
     1. Let _set_ be the *this* value.
-    1. If Type(_set_) is not Object, throw a *TypeError* exception.
```

Comment:
I assumed that check was here because of the internal slot check and in [`union`](https://github.com/tc39/proposal-set-methods/blob/ced709ed0c9c95ed08bab711408bb7842e0a4c74/spec/set-prototype-union.html#L5) etc because those usages immediately precede a call to [`SpeciesConstructor`](https://tc39.github.io/ecma262/#sec-speciesconstructor), which requires its input to be an object. Neither of those apply. It's omitted in this PR because I was copying the other explicit uses of the iteration protocol in e.g. [`isSupersetOf`](https://github.com/tc39/proposal-set-methods/blob/ced709ed0c9c95ed08bab711408bb7842e0a4c74/spec/set-prototype-is-superset-of.html#L8), which do not include the check.

Anyway, I don't really have an opinion on this either way; I'm happy to put it back if @gsathya has a preference for it being there. I think the only practical effect of having the check is that `Set.prototype.isSubsetOf('ab', ['a', 'b'])` would throw instead of returning the right answer, which is not a case I'm really concerned about (though I see no reason for it not to work).

Reply:
`Set.prototype.isSubsetOf.call('ab', ['a', 'b'])` is an interesting use case. Maybe for this case makes sense update `union` and the rest methods from this check to `ToObject` conversion?

Reply:
I'm really not interested in making this work for all kinds of cases. I like the current version.